app/model/repository/Pipelines.php

```diff
@@ -17,14 +21,46 @@ public function __construct(EntityManager $em) {
     parent::__construct($em, Pipeline::class);
   }
 
+  // Known order by commands and their translation to Doctrine column names.
+  private static $knownOrderBy = [
+    'name' =>      [ 'p.name' ],
+    'createdAt' => [ 'p.createdAt' ],
+  ];
+
 
   /**
-   * Search pipelines names based on given string.
-   * @param string|null $search
+   * Search pipelines based on parameters (filters and orderBy) in pagination config.
+   * The result is not sliced as pipelines must be filtered by ACL first.
+   * @param Pagination $pagination Pagination configuration object.
    * @return Pipeline[]
    */
-  public function searchByName(?string $search): array {
-    return $this->searchBy(["name"], $search);
+  public function getPreparedForPagination(Pagination $pagination): array {
+    $qb = $this->createQueryBuilder('p'); // takes care of softdelete cases
+
+    // Set filters ...
+    if ($pagination->hasFilter("search")) {
+      $search = trim($pagination->getFilter("search"));
+      if (!$search) {
+        throw new InvalidArgumentException("filter", "search query value is empty");
+      }
+
+      $qb->andWhere($qb->expr()->like("p.name", $qb->expr()->literal('%' . $search . '%')));
+    }
+
+    // Add order by
+    if ($pagination->getOrderBy() && !empty(self::$knownOrderBy[$pagination->getOrderBy()])) {
```

Comment:
If I am not mistaken, this piece of code is the same for `Users`, maybe a helper function would be on point? ... the same goes for detection of search filter?

Reply:
OK. I will give it a serious thought.